tools/export.py

```diff
@@ -22,51 +31,31 @@
 import numpy as np
 
 
-def export(name, task='rec', local_ckpt_path="", save_dir=""):
+def export(name, data_shape, local_ckpt_path="", save_dir=""):
     ms.set_context(mode=ms.GRAPH_MODE) #, device_target='Ascend')
     if local_ckpt_path:
         net = build_model(name, pretrained=False, ckpt_load_path=local_ckpt_path)
     else:
         net = build_model(name, pretrained=True)
     net.set_train(False)
 
-    # TODO: extend input shapes for more models
-    if task == 'rec':
-        c, h, w = 3, 32, 100
-    else:
-        c, h, w = 3, 736, 1280
-
-    bs = 1
+    h, w = data_shape
+    bs, c = 1, 3
     x = ms.Tensor(np.ones([bs, c, h, w]), dtype=ms.float32)
 
     output_path = os.path.join(save_dir, name) + '.mindir'
     ms.export(net, x, file_name=output_path, file_format='MINDIR')
 
-    print(f'=> Finish exporting {name} to {output_path}')
-
-
-def str2bool(v):
-    if isinstance(v, bool):
-        return v
-    if v.lower() in ("yes", "true", "1"):
-        return True
-    elif v.lower() in ("no", "false", "0"):
-        return False
-    else:
-        raise argparse.ArgumentTypeError("Boolean value expected.")
+    print(f'=> Finish exporting {name} to {os.path.realpath(output_path)}. The data shape [H, W] is {data_shape}')
 
 
 def check_args(args):
-    if args.local_ckpt_path:  # load local ckpt
-        if not args.model_name:
-            raise ValueError("Arg 'model_name' is empty. Please set 'model_name' if 'local_ckpt_path' is not None.")
-        if args.model_name not in list_models():
-            raise ValueError(f"Invalid 'model_name': {args.model_name}. 'model_name' must be one of names in {list_models()}.")
-        if not os.path.isfile(args.local_ckpt_path):
-            raise ValueError(f"No such ckpt file in this path: {args.local_ckpt_path}.")
-    else:  # download online ckpt
-        if args.model_name and args.model_name not in list_models():
-            raise ValueError(f"Invalid 'model_name': {args.model_name}. 'model_name' must be empty or one of names in {list_models()}.")
+    if args.model_name not in list_models():
+        raise ValueError(f"Invalid arg 'model_name': {args.model_name}. 'model_name' must be empty or one of names in {list_models()}.")
+    if args.local_ckpt_path and not os.path.isfile(args.local_ckpt_path):
+        raise ValueError(f"Local ckpt file {args.local_ckpt_path} does not exist. Please check arg 'local_ckpt_path'.")
+    if args.save_dir and not os.path.isdir(args.save_dir):
+        raise ValueError(f"Directory {args.save_dir} does not exist. Please check arg 'save_dir'.")
```

Comment:
Can we just create a directory if it doesn't exist?

Reply:
updated